### PR TITLE
LPS-190069 ISO3166 Country Codes are not updated with latest ISO list

### PR DESCRIPTION
**Pull request by ces-locpham**
Hi team, 
Please help me review this ticket.
https://liferay.atlassian.net/browse/LPS-190069
It is reviewed at https://github.com/liferay-commerce/liferay-portal/pull/4359
Many thanks.
cc: @wanderlast

### CODE REVIEW

File: modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/GB.json [hunk 1]
```diff
@@ -92,7 +92,7 @@
 		"regionCode": "BEN"
 	},
 	{
-		"name": "Bridgend",
+		"name": "Bridgend [Pen-y-bont ar Ogwr GB-POG]",
```
Comment (smottal) [inline modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/GB.json:95]: Are these names with ```[code] ``` correct?

File: modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/FR.json [hunk 2]
```diff
@@ -320,12 +324,12 @@
 		"regionCode": "53"
 	},
 	{
-		"name": "Mayotte (dept)",
+		"name": "Mayotte",
 		"regionCode": "976"
 	},
 	{
-		"name": "Mayotte (region)",
-		"regionCode": "YT"
```
Comment (smottal) [inline modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/FR.json:328]: What about the already existing addresses that are using this regioneCode?

File: modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/FR.json [hunk 3]
```diff
@@ -377,7 +381,7 @@
 	},
 	{
 		"name": "Paris",
-		"regionCode": "75"
```
Comment (smottal) [inline modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/FR.json:380]: Same for this

File: modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/GB.json [hunk 4]
```diff
@@ -548,8 +556,8 @@
 		"regionCode": "NYK"
 	},
 	{
-		"name": "Northamptonshire",
-		"regionCode": "NTH"
```
Comment (smottal) [inline modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/GB.json:552]: What about the already existing addresses that are using this regioneCode?

File: modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/JP.json [hunk 5]
```diff
@@ -4,328 +4,328 @@
 			"ja_JP": "愛知県"
 		},
 		"name": "Aichi",
-		"regionCode": 23.0
+		"regionCode": 23
```
Comment (smottal) [inline modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/JP.json:7]: In all the other files the regioneCode is a string. Please update also this one

File: modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/KR.json [hunk 6]
```diff
@@ -1,70 +1,70 @@
 [
 	{
 		"name": "Busan-gwangyeoksi",
-		"regionCode": 26.0
+		"regionCode": "26"
```
Comment (smottal) [inline modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/KR.json:4]: I think that also in this case we need some type of update

File: modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/MX.json [hunk 7]
```diff
@@ -25,7 +25,7 @@
 	},
 	{
 		"name": "Ciudad de México",
-		"regionCode": "DIF"
+		"regionCode": "CMX"
```
Comment (smottal) [inline modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/MX.json:28]: Upgrade process?

File: modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/NA.json [hunk 8]
```diff
@@ -0,0 +1,58 @@
+[
+	{
+		"name": "//Karas",
```
Comment (smottal) [inline modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/NA.json:3]: Is it correct?

File: modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/NO.json [hunk 9]
```diff
@@ -1,54 +1,54 @@
 [
 	{
 		"name": "Agder",
-		"regionCode": 42.0
+		"regionCode": "42"
```
Comment (smottal) [inline modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/NO.json:4]: Upgrade process?

File: modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/NP.json [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+	{
+		"name": "Lumbini",
+		"regionCode": "P5"
+	},
+	{
+		"name": "Madhesh",
+		"regionCode": "P2"
+	},
+	{
+		"name": "Province 1",
```
Comment (smottal) [inline modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/NP.json:23]: Is it correct?

File: modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/PL.json [hunk 11]
```diff
@@ -1,66 +1,66 @@
 [
 	{
 		"name": "Dolnośląskie",
-		"regionCode": 2.0
+		"regionCode": "2"
```
Comment (smottal) [inline modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/PL.json:4]: Upgrade?

File: modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/SA.json [hunk 12]
```diff
@@ -1,54 +1,54 @@
 [
 	{
-		"name": "'Asir",
-		"regionCode": 14.0
+		"name": "'Asīr",
+		"regionCode": "14"
```
Comment (smottal) [inline modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/SA.json:4]: Upgrade?

File: modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/TR.json [hunk 13]
```diff
@@ -1,326 +1,326 @@
 [
 	{
 		"name": "Adana",
-		"regionCode": 1.0
+		"regionCode": "1"
```
Comment (smottal) [inline modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/TR.json:4]: Upgrade?

**Review by smottal — CHANGES_REQUESTED** [review]
Hi @locpham97 , could you please have a look to my comments?

File: modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/FR.json (hunk 2 above)
Comment (locpham97) [reply modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/FR.json:328]: I also follow the ISO standard to update the region JSON

File: modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/NA.json (hunk 8 above)
Comment (locpham97) [reply modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/NA.json:3]: Yes, I follow here https://www.iso.org/obp/ui/#iso:code:3166:NA

**Comment by locpham97** [comment]
I updated it following your guidance.
Could you check it again
@smottal

File: modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/GB.json (hunk 4 above)
Comment (locpham97) [reply modules/apps/address/address-impl/src/main/resources/com/liferay/address/dependencies/regions/GB.json:552]: It is separated two unary regions [West Northamptonshire](https://en.wikipedia.org/wiki/West_Northamptonshire)
[North Northamptonshire](https://en.wikipedia.org/wiki/North_Northamptonshire)

**Comment by smottal** [comment]
Hi @locpham97, I would like to keep the values as uniform as possible (so all strings because in some cases we have to manage chars) and, because an upgrade process is needed as discussed in jira, is better to uniform all the numbers (or all decimals or all integers).

**Comment by locpham97** [comment]
Hello @smottal,

Regarding the upgrade process for the region code update, I've reviewed the address-impl module and found the repopulate OSGi command. This command is specifically designed to reinitialize the country region. Therefore, I propose utilizing this OSGi command to re-update Liferay data. If necessary, the OSGi command can be modified to tailor the region information update.

However, if you don't accept the approach, I will proceed with creating an upgrade process specifically for this ticket.

I would be grateful if you could review the proposed approach and provide your preferred method for handling the region code update. Your insights are highly valued.

Thank you for your time and consideration.

Sincerely,
Loc

**Comment by locpham97** [comment]
ci:test:sf

**Comment by locpham97** [comment]
ci:test:stable

**Comment by locpham97** [comment]
ci:test:relevant

**Comment by locpham97** [comment]
Hi @smottal , 
I updated it PR. However, I also consider how we can handle the deleted regions. Can I deactive the deleted regions ?

**Comment by smottal** [comment]
Resent here: https://github.com/liferay-commerce/liferay-portal/pull/4589

**Pull request closed without merging**